extensions/eclipse-che-theia-preferences-provider-extension/src/browser/prefs-provider.ts

```diff
@@ -32,19 +32,37 @@ export class PreferencesProvider implements FrontendApplicationContribution {
 
   private async getPluginsProperties(): Promise<[string, string][]> {
     const devfile = await this.devfileService.get();
+    const componentStatuses = await this.devfileService.getComponentStatuses();
     const components = devfile.components;
     if (!components) {
       throw new TypeError('Can\'t get "components" of current workspace "devfile" section.');
     }
 
-    return components
-      .map(component => {
-        if (component.plugin) {
-          return component.plugin.preferences || {};
-        } else {
-          return {};
+    const devfilePluginPreferences = components.map(component => {
+      if (component.plugin) {
+        return component.plugin.preferences || {};
+      }
+      return {};
+    });
+
+    const componentStatusPreferences = componentStatuses.map(componentStatus => {
+      // check if env var is matching
+      if (componentStatus?.env) {
+        const preferencesEnv = componentStatus.env.find(env => env.name === 'CHE_THEIA_SIDECAR_PREFERENCES');
```

Comment:
So the env variable is not used as an env variable.

Reply:
it's a marker yes for che-server usecase.

For DevWorkspace usecase, preferences will be read directly from che-theia-plugins.yaml files